controllers/topolvm_controller.go

```diff
@@ -0,0 +1,325 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	lvmv1alpha1 "github.com/red-hat-storage/lvm-operator/api/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	cutil "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	controllerName = "topolvm-controller"
+)
+
+type topolvmController struct{}
+
+// topolvmController unit satisfies resourceManager interface
+var _ resourceManager = topolvmController{}
+
+func (c topolvmController) getName() string {
+	return controllerName
+}
+
+//+kubebuilder:rbac:groups=apps,resources=deployments,verbs=create;update;delete;get;list;watch
+
+func (c topolvmController) ensureCreated(r *LVMClusterReconciler, ctx context.Context, lvmCluster *lvmv1alpha1.LVMCluster) error {
+
+	// get the desired state of topolvm controller deployment
+	controllerDeployment := getControllerDeployment(lvmCluster)
+
+	result, err := cutil.CreateOrUpdate(ctx, r.Client, controllerDeployment, func() error { return nil })
```

Comment:
The `MutateFn` (last parameter of this function) is empty.  So it will never update the deployment. I don't think that's what we want here.

Reply:
or maybe I'm missing something here.

Reply:
> it will never update the deployment
- I don't think so, we don't want to mutate the deployment, if there's an upgrade scenario then controller-util will compare the spec's and update it (if you meant deployment update)
- When LVMCluster CR is changed then also deployment doesn't need to be updated (if you meant in CR context)

Reply:
thanks for the clarification.